index.test.js

```diff
@@ -0,0 +1,233 @@
+
+    it('should return false when inputs differ by characters that do matter', () => {
+      const instance = new Rubric()
+        .doesntMatter(Rubric.en.COMMON_PUNCTUATION)
+        .matters('-');
+      const inputs = [['fire-fly light', 'firefly light']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+
+  describe('match (capitalization doesnt matter)', () => {
+    it('should return true when inputs differ solely by capitalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.CAPITALIZATION);
+      const inputs = [['us', 'US']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: true });
+      });
+    });
+
+    it('should return false when inputs differ other than by capitalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.CAPITALIZATION);
+      const inputs = [['us', 'usa']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+
+  describe('match (whitespace doesnt matter)', () => {
+    it('should return true when inputs differ solely by shape of whitespaces', () => {
+      const instance = new Rubric().doesntMatter(Rubric.WHITESPACE_DIFFERENCES);
+      const inputs = [['the us of a', 'the	us   of\u2028\u2029a']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: true });
+      });
+    });
+
+    it('should return false when inputs differ other than by shape of whitespaces', () => {
+      const instance = new Rubric().doesntMatter(Rubric.WHITESPACE_DIFFERENCES);
+      const inputs = [['the us of a', 'The	us   of\u2028\u2029a']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+
+  describe('match (unicode normalization doesnt matter)', () => {
+    it('should return true when inputs differ solely by unicode normalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.UNICODE_NORMALIZATION);
+      // composed and decomposed forms of é
+      const inputs = [['\u00e9', '\u0065\u0301']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: true });
+      });
+    });
+
+    it('should return false when inputs differ other than by unicode normalization', () => {
+      const instance = new Rubric().doesntMatter(Rubric.UNICODE_NORMALIZATION);
+      // lowercase n \u006e and uppercase N \u004e with combining tilde \u0303
+      const inputs = [['\u006e\u0303', '\u004e\u0303']];
+      inputs.forEach(([s1, s2]) => {
+        expect(instance.match(s1, s2)).toEqual({ isMatch: false });
+      });
+    });
+  });
+});
+
+describe('Real-world usage', () => {
+  describe('es rubric', () => {
+    const esRubric = new Rubric()
+      .doesntMatter(Rubric.UNICODE_NORMALIZATION)
```

Comment:
I have implemented, or seen implemented, these five lines more or less, imperatively, piecemeal/partially, with slightly different behaviors, probably a dozen times over the years.